src/content/api/parser.md

```diff
@@ -37,279 +38,477 @@ as such:
+export class ClassName {}
+```
 
 
 ### exportExpression
 
 `SyncBailHook`
 
-...
+Called for every `export` statement exporting an expression e.g.`export default expression;`.
 
-Parameters: `statement` `declaration`
+- Callback Parameters: `statement` `declaration`
 
 
 ### exportSpecifier
 
 `SyncBailHook`
 
-...
+Called for every specifier of every `export` statement.
 
-Parameters: `statement` `identifierName` `exportName` `index`
+- Callback Parameters: `statement` `identifierName` `exportName` `index`
 
 
 ### exportImportSpecifier
 
 `SyncBailHook`
 
-...
+Called for every specifier of every `export`-import statement.
 
-Parameters: `statement` `source` `identifierName` `exportName` `index`
+- Callback Parameters: `statement` `source` `identifierName` `exportName` `index`
 
 
 ### varDeclaration
 
 `SyncBailHook`
 
-...
+Called when parsing a variable declaration.
 
-Parameters: `declaration`
+- Callbak Parameters: `declaration`
 
 
 ### varDeclarationLet
 
 `SyncBailHook`
 
-...
+Called when parsing a variable declaration defined using `let`
 
-Parameters: `declaration`
+- Callbak Parameters: `declaration`
 
 
 ### varDeclarationConst
 
 `SyncBailHook`
 
-...
+Called when parsing a variable declaration defined using `const`
 
-Parameters: `declaration`
+- Callbak Parameters: `declaration`
 
 
 ### varDeclarationVar
 
 `SyncBailHook`
 
-...
+Called when parsing a variable declaration defined using `var`
 
-Parameters: `declaration`
+- Callbak Parameters: `declaration`
 
 
 ### canRename
 
 `SyncBailHook`
 
-...
+Triggered before renaming an identifier to determine if the renaming is allowed. This is usually used together with the `rename` hook.
+
+- Hook Parameters: `identifier`
+- Callback Parameters: `expression`
 
-Parameters: `initExpression`
+```js
+var a = b;
+
+parser.hooks.canRename.for('b').tap('MyPlugin', expression => {
+  // returning true allows renaming
+  return true;
+});
+```
 
 
 ### rename
 
 `SyncBailHook`
 
-...
+Triggered when renaming to get the new identifier. This hook will be called only if `canRename` returns `true`.
+
+- Hook Parameters: `identifier`
+- Callback Parameters: `expression`
 
-Parameters: `initExpression`
+```js
+var a = b;
+
+parser.hooks.rename.for('b').tap('MyPlugin', expression => {});
+```
 
 
 ### assigned
 
 `SyncBailHook`
 
-...
+Called when parsing an `AssignmentExpression` before parsing the assigned expression.
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+- Callback Parameters: `expression`
+
+```js
+a += b;
+
+parser.hooks.assigned.for('a').tap('MyPlugin', expression => {
+  /* this is called before parsing b */
+});
+```
 
 
 ### assign
 
 `SyncBailHook`
 
-...
+Called when parsing an `AssignmentExpression` before parsing the assign expression.
 
-Parameters: `expression`
+- Hook Parameters: `identifier`
+- Callback Parameters: `expression`
+
+```js
+a += b;
+
+parser.hooks.assigned.for('a').tap('MyPlugin', expression => {
+  /* this is called before parsing a */
```

Comment:
can we use two forward slashes for all the single line comments in js code example blocks for the sake of consistency of the document?